pkg/ipmi/blob_transfer.go

```diff
@@ -0,0 +1,409 @@
+// Copyright 2020 the u-root Authors. All rights reserved
```

Comment:
I'm starting to wonder about the structure of all of this.

If we keep going like this, eventually, `pkg/ipmi` is going to grow to be a kitchen sink of all kinds of OEM commands and implementations.

Do you think it'd be possible to make `pkg/ipmi` the generic ipmi device (and have it implement only generic IPMI commands), and have a `pkg/ipmi/wiwynn` for Wiwynn-specific things, and `pkg/ipmi/blobs` for this blob protocol?

That'd probably mean exporting something like Req and Recv (perhaps call them Request and Response, spell them out), and exposing the SendRecv function. 

cc @johnnylinwiwynn @pmazzini @insomniacslk

Reply:
I restructured it the way you described, so there's `pkg/ipmi`, `pkg/ipmi/blobs`, and `pkg/ipmi/wiwynn`

To make that work, I exposed most of the commands in `ipmi.go`, `SendRecv` and the `msg` struct, but hopefully that makes the interface a little easier to use - instead of having to deal with req, you just call `SendRecv` with a msg now, which seems to be what most of these functions were doing anyway. 

But since most of those were methods defined on the IPMI type and thats not doable anymore for the Wiwynn functions with IPMI being non-local, I changed those to take in `*ipmi.IPMI` as a parameter